### ISSUE TITLE
Nuxt V2 - $model undefined

### ISSUE THREAD

**Issue by lsbyerley** [vuelidate/vuelidate#1099]
**Describe the bug**
A `Cannot read properties of undefined (reading '$model')` error when using Vuelidate 2.x with Nuxt V2

I'm not sure why the $model would be undefined. Does anything seem wrong in the component?

**Additional context**
`"nuxt": "2.15.8"`

```
<template>
    <div class="container mx-auto">
      <form
        class="w-full max-w-md p-6 mx-auto text-sm text-gray-600 rounded"
        ref="piForm"
        @submit.prevent="submitForm"
      >
        <div class="relative mb-4 rounded shadow appearance-none label-floating">
          <input
            class="w-full px-3 py-2 leading-normal rounded"
            id="parentPhone"
            type="text"
            placeholder="Parent Phone"
            v-model="v$.parentPhone.$model" // ERROR POINTS TO THIS LINE
          />
          <label
            class="absolute top-0 left-0 block w-full px-3 py-2 leading-normal text-left"
            for="parentPhone"
          >
            Parent Phone
          </label>
          <span class="text-red-600" v-if="errors && v$.parentPhone.$invalid"
            >Valid phone is required!</span
          >
        </div>
      </form>
    </div>
</template>

<script>
import useVuelidate from '@vuelidate/core';
import { required, email, numeric } from '@vuelidate/validators';

export default {
  setup() {
    return { v$: useVuelidate() };
  },
  data() {
    return {
      parentPhone: '',
    };
  },
  validations() {
    return {
      parentPhone: {
        required,
        numeric,
        minLength: minLength(10),
      },
    };
  },
  methods: {
    async submitForm() {
      const isFormCorrect = await this.v$.$validate();

      if (!isFormCorrect) {
        return;
      }

      //submit form
    }
  }
};
</script>
```

**Expected behavior**
no error

**Comment by aKzenT** [comment]
I'm having the same problem after upgrading from 0.x to 2.x.

**Comment by ahsath** [comment]
Possibly related #1101

**Comment by xeno** [comment]
I don't think this will help with your situation, but leaving it here for other people who may find this thread.

I was just having this issue while trying an async validator in nuxt3. My issue was my rules weren't configured properly. I don't know why that caused a $model error, but it did.

**Comment by lehoaibaokg** [comment]
I think the HTML renders before the validation is defined. You can try to catch it with a flag default is false. When the screen is mounted then set the flag to true, if the flag is true then the page will be rendered.
E.g:
`<div class="container mx-auto" v-if="flag">`

```
data: function () {
    return {
        flag: false
    }
}
```

```
mounted: function () {
    this.flag = true;
}
```

**Comment by IGx89** [comment]
Just ran into this myself with a similar setup (Vue 2.7.14, Vuelidate 2, vue-server-renderer, Options API). The root cause is that Vue 2 SSR just runs the setup, beforeCreate and created lifecycle hooks ([reference](https://v2.ssr.vuejs.org/guide/universal.html#component-lifecycle-hooks)), skipping beforeMount and later hooks. Vuelidate 2's useVuelidate runs the validations() function in an onBeforeMount hook when the no-arg Options API overload is called, so it never gets loaded during SSR.

The solution is to use the Composition API (which initializes the validations in setup instead of beforeMount) or to adjust your template to not use v$ during SSR scenarios (such as the v-if flag suggested before, or through optional chaining like `v$.myModel?.$error`). I've decided to use optional chaining and that's working well so far.

Whether this is a vuelidate bug or by design, dunno. I could definitely see an argument for validations not being relevant to SSR, though it is a bit annoying. Yet another reason to move to Composition API I suppose :p
